controller/resource.go

```diff
@@ -174,17 +176,57 @@ func (t *Task) Upsert() error {
 	}
 
 	// Otherwise, create one.
-	t.Log.Info("creating service token", "id", serviceName)
 	err = t.updateServiceToken(serviceName, secretName)
 	if err != nil {
 		return fmt.Errorf("updating service token: %w", err)
 	}
-	t.Log.Info("service token created successfully", "id", serviceName)
 
 	return nil
 }
 
 func (t *Task) Delete() error {
+	if !t.isMeshTask() {
+		t.Log.Info("skipping non-mesh task", "id", *t.Task.TaskDefinitionArn)
+		return nil
+	}
+
+	tokenList, _, err := t.ConsulClient.ACL().TokenList(nil)
```

Comment:
Would this list all tokens in Consul? We are potentially doing this multiple times per reconcile if they churn a lot of tasks at once.

Could we move this out of `Delete`, so that we are sure to call TokenList at most once per `reconcile`?

Reply:
Yeah, we can revisit later!